### PR TITLE
graceful grpc server shutdown

### PR DESCRIPTION
**Pull request by vangheem**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb_node/tests/common/node_services.rs [hunk 1]
```diff
@@ -321,18 +344,7 @@ impl NodeFixture {
...
-        }
-        if let Some(task) = self.writer_server_task.take() {
-            task.abort();
-        };
-        if let Some(task) = self.secondary_reader_server_task.take() {
-            task.abort();
-        };
-        if let Some(task) = self.secondary_writer_server_task.take() {
-            task.abort();
-        };
```
Comment (hermeGarcia) [inline nucliadb_node/tests/common/node_services.rs:336]: Much better!

File: nucliadb_node/src/replication/replicator.rs [hunk 2]
```diff
@@ -360,15 +370,21 @@ pub async fn connect_to_primary_and_replicate_forever(
...
             secondary_id.clone(),
+            Arc::clone(&shutdown_notified),
         )
         .await;
-        if result.is_err() {
+
+        if let Err(_) = result {
             error!(
                 "Error happened during replication. Will retry: {:?}",
                 result
```
Comment (hermeGarcia) [inline nucliadb_node/src/replication/replicator.rs:390]: Consider the following to get rid of the clippy warning: 
```rust
        if result.is_ok() {
            return Ok(());
        }

        error!(
            "Error happened during replication. Will retry: {:?}",
            result
        );
        tokio::time::sleep(std::time::Duration::from_secs(
            settings.replication_delay_seconds(),
        ))
        .await;
```

**Review by hermeGarcia — CHANGES_REQUESTED** [review]
Just a change to make clippy happy!

File: nucliadb_node/src/replication/replicator.rs (hunk 2 above)
Comment (vangheem) [reply nucliadb_node/src/replication/replicator.rs]: it's funny, I was trying to be more rustic!

**Review by hermeGarcia — APPROVED** [review]

**Pull request merged**